src/masternodes/mn_checks.cpp

```diff
@@ -2224,6 +2224,8 @@ Res CPoolSwap::ExecuteSwap(CCustomCSView& view, std::vector<DCT_ID> poolIDs) {
         poolPrice = obj.maxPrice;
     }
 
+    CCustomCSView intermediateView(view);
```

Comment:
Then use only that view in the loop otherwise SubBalance may fails due to distinct views. So just use intermediate view inside the loop

Reply:
Good point. Thanks for spotting that.